autodiff/forward/forward.hpp

```diff
@@ -1487,6 +1493,15 @@ constexpr void apply(Dual<T, G>& self, AbsOp)
     self.grad *= aux / self.val;
 }
 
+template<typename T, typename G>
+constexpr void apply(Dual<T, G>& self, ErfOp)
+{
+    constexpr double pi = 3.1415926535897932384626433832795029;
+    const T aux = self.val;
+    self.val = erf(aux);
+    self.grad *= 2.0 * exp(-aux*aux)/sqrt(pi);
```

Comment:
Let's precompute `sqrt(pi)`. Unfortunately, I think `std::sqrt` is not `constexpr`, so we set `sqrt_pi` instead of `pi`.

Reply:
Sorry, are you suggesting: 
``constexpr double sqrt_pi = std::sqrt(3.1415926535897932384626433832795029);``